CMakeLists.txt

```diff
@@ -292,27 +292,27 @@ if(BUILD_EXAMPLES AND NOT WIN32)
     target_compile_definitions(hello_module PRIVATE ${qjs_defines})
     target_link_libraries(hello_module ${qjs_libs})
 
-    if(NOT WIN32)
-        add_library(fib MODULE examples/fib.c)
-        set_target_properties(fib PROPERTIES
-            PREFIX ""
-            C_VISIBILITY_PRESET default
-        )
-        target_compile_definitions(fib PRIVATE JS_SHARED_LIBRARY)
-        if(APPLE)
-            target_link_options(fib PRIVATE -undefined dynamic_lookup)
-        endif()
+    add_library(fib MODULE examples/fib.c)
+    set_target_properties(fib PROPERTIES
+        PREFIX ""
+        C_VISIBILITY_PRESET default
+    )
+    target_compile_definitions(fib PRIVATE JS_SHARED_LIBRARY)
+    target_link_libraries(fib ${qjs_libs})
```

Comment:
Ah this is what causes the linking error in Ubuntu, actually. On unix we don't need to link the module with the library.

Reply:
Maybe it should be?
``` diff
-target_link_libraries(fib ${qjs_libs})
+if(WIN32)
+      target_link_libraries(fib ${qjs_libs})
+endif()
```

Reply:
I think so yeah

Reply:
@saghul 
[change link](https://github.com/quickjs-ng/quickjs/pull/490/commits/096ada0a027da304788b2c1f4756bfbf2948e538)